stp/stp.zkasm

```diff
@@ -0,0 +1,163 @@
+;; The STP module performs upfront gas calculations for CALL and
+;; CREATE instructions.  The logic of this module is tricky in places,
+;; and is largely determined by the Yellow Paper.  For any call to
+;; stp, the "inst" argument indicates which of these instructions is
+;; being computed:
+;;
+;; CREATE                           0xF0 (240)
+;; CALL                             0xF1 (241)
+;; CALLCODE                         0xF2 (242)
+;; DELEGATECALL                     0xF4 (244)
+;; CREATE2                          0xF5 (245)
+;; STATICCALL                       0xFA (250)
+;;
+;; G_txcreate                       32000
+;; G_callvalue                      9000
+;; G_callstipend                    2300
+;; G_new_account                    25000
+;; G_warmaccess                     100
+;; G_coldaccountaccess             2600
+;;
+;; Here, GAS_ACTUAL holds the amount of gas available to the execution
+;; context just prior to the instruction.  GAS_MXP holds the gas cost
+;; for any memory expansion which maybe required for holding the call
+;; / return data, etc.  GAS holds the amount of gas provided on the
+;; stack as parameters to the call, whilst VALUE holds the amount in
+;; wei provided as a parameter to the CALL/CALLCODE instruction.
+;; Finally, EXISTS indicates for a CALL instruction whether target
+;; account exists already, and WARM indicates whether the target
+;; account for a call instruction is warm (or not).
+fn stp(INST=0xf0 u8, GAS_ACTUAL u64, GAS_MXP u64, GAS u256, VALUE u256, EXISTS u1, WARM u1) -> (OOGX=1 u1, GAS_UPFRONT=32000 u64, GAS_OOP u64, GAS_STIPEND u64)
```

Comment:
I remember you mentioned somewhere else how to interpret expressions such as `INST=0xf0 u8` or `GAS_UPFRONT=32000 u64` could you remind me of the interpretation ?